packages/deposit/lib/src/adapters/memory_deposit_adapter.dart

```diff
@@ -83,7 +101,18 @@ class MemoryDepositAdapter extends DepositAdapter<int> {
     String primaryColumn,
     Map<String, dynamic> data,
   ) async {
-    _ref(table).removeAt((data[primaryColumn] as int) - 1);
+    _ref(table).removeWhere(
+      (d) => d[primaryColumn] == data[primaryColumn],
+    );
+  }
+
+  @override
+  Future<void> removeAll(
+    String table,
+    String primaryColumn,
+    List<Map<String, dynamic>> data,
+  ) async {
```

Comment:
```suggestion
  ) {
```

Reply:
This one has to stay `async` cause the `Future.wait` is not returned, but you do want to wait until it is finished